packages/twenty-server/src/engine/metadata-modules/object-metadata/utils/build-workspace-migrations-for-remote-object.ts

```diff
@@ -7,21 +9,60 @@ import {
 import { computeCustomName } from 'src/engine/utils/compute-custom-name.util';
 import { computeObjectTargetTable } from 'src/engine/utils/compute-object-target-table.util';
 
-const buildCommentForRemoteObjectForeignKey = (
+const buildCommentForRemoteObjectForeignKey = async (
   localObjectMetadataName: string,
   remoteObjectMetadataName: string,
   schema: string,
-): string =>
-  `@graphql({"totalCount":{"enabled": true},"foreign_keys":[{"local_name":"${localObjectMetadataName}Collection","local_columns":["${remoteObjectMetadataName}Id"],"foreign_name":"${remoteObjectMetadataName}","foreign_schema":"${schema}","foreign_table":"${remoteObjectMetadataName}","foreign_columns":["id"]}]})`;
+  workspaceDataSource: DataSource | undefined,
+): Promise<string> => {
+  const existingComment = await workspaceDataSource?.query(
+    `SELECT col_description('${schema}."${localObjectMetadataName}"'::regclass, 0)`,
+  );
+
+  if (!existingComment[0]?.col_description) {
+    return `@graphql({"totalCount":{"enabled": true},"foreign_keys":[{"local_name":"${localObjectMetadataName}Collection","local_columns":["${remoteObjectMetadataName}Id"],"foreign_name":"${remoteObjectMetadataName}","foreign_schema":"${schema}","foreign_table":"${remoteObjectMetadataName}","foreign_columns":["id"]}]})`;
+  }
+
+  const commentWithoutGraphQL = existingComment[0].col_description
+    .replace('@graphql(', '')
+    .replace(')', '');
+  const parsedComment = JSON.parse(commentWithoutGraphQL);
+
+  if (parsedComment.foreign_keys) {
```

Comment:
I think you can make this more concise and avoid repetition there 👍

Reply:
Indeed, thanks. Updated!